core/src/test/scala/org/apache/spark/JobCancellationSuite.scala

```diff
@@ -234,7 +234,7 @@ class JobCancellationSuite extends SparkFunSuite with Matchers with BeforeAndAft
     val jobA = Future {
       sc.setJobGroup("jobA", "this is a job to be cancelled", interruptOnCancel = true)
       sc.parallelize(1 to 10000, 2).map { i =>
-        while (true) { }
+        while (true) { Thread.sleep(100) }
```

Comment:
If we are entangled in this Infinite loop, we can add an `AtomicBoolean` status variable and update it to `false` in  finally block to jump out of the loop